languages/javascript/reference/keywords/export.md

```diff
@@ -0,0 +1,29 @@
+# `export`
+
+The `export` keyword is used to define elements of the current file, that can be imported into other files using the [`import`][concept-import] keyword.
```

Comment:
```suggestion
The `export` keyword is used to define which elements in the current file will be importable into other files using the [`import`][keyword-import] keyword.
```

What do you think of this wording? If you like it better, don't apply the above.

Reply:
Perhaps you also want to say that a file becomes a module once there is a single export?

Reply:
Does it have to have an export to be a module? I always understood all files to be a module.

Reply:
> In the Node.js module system, each file is treated as a separate module. 
> https://nodejs.org/api/modules.html#modules_modules

> Adding JavaScript modules to the web platform
> https://blog.whatwg.org/js-modules

> ECMAScript Language: Scripts and Modules
> https://tc39.es/ecma262/#sec-ecmascript-language-scripts-and-modules

As far as I can tell, ECMAScript *does not* dictate _when_ something is a module, but it does tell us that keywords such as `import` and `export` are only available from a "module-context".  

- The browser platform, however, is pretty clear about this. A `<script>` becomes a module, when it has `type="module"` and ES6 style specifiers such as `import` and `export` are *only* available in such modules.
- Node is also pretty clear about this. Each _file_ is a module, by default. That does not mean that all tools will interpret it the same way. For example, `typescript` or `eslint` (one of the two, I don't know which one of the top of my head) only marks a _file_ as a _module_ once there is a single `import` *or* `export` in the file.

But the fact that I had to look this up is pretty interesting. We should at least mention that the `import` and `export` keywords are only available in a module-context which means files when in a node-context, and `<script type="module">` in a browser-context.

Reply:
Here is an amazing resource to include (in the files):

https://exploringjs.com/impatient-js/ch_modules.html